Turn string constraints into a struct

romainbrenguier opened the pull request:
This used to be a class. But we can have it a a simple struct instead, and the code looks simpler.

Commit:
Move string_constraint functions out of string_refinement

romainbrenguier committed:
Drop string constraint:exprt inheritance

romainbrenguier committed:
Rename premise in index_guard in string constraint

LAJW reviewed (approved):
Yes, please, as long as it compiles.

romainbrenguier committed:
Refactor find_qvar

Use standard algorithm instead of defining a visitor class

romainbrenguier committed:
Drop doxygen link to string_constraintt

Doxygen gives warning on this link command, because it is not able to
resolve it.

romainbrenguier committed:
Fix documentation for is_valid_string_constraint

Make doxygen stop complaining about missing fields.

romainbrenguier committed:
Remove java dependency

Lower bound should not use java_type

romainbrenguier commented:
This is now outdated and similar changes were made in another PR which is now merged.
